excel.go

```diff
@@ -0,0 +1,245 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/xuri/excelize/v2"
+)
+
+func getHeaderToXCoord(headerRow []string) map[string]int {
+	headerToXCoord := make(map[string]int, len(headerRow))
+	for i, cell := range headerRow {
+		headerToXCoord[cell] = i
+	}
+	return headerToXCoord
+}
+
+func getMACFinUsers(f *excelize.File, input *Input) (map[string]string, error) {
+	rows, err := f.GetRows(input.SheetName)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting rows from %s in %s: %s", input.SheetName, input.Filename, err)
+	}
+
+	users := make(map[string]string)
+
+	headerToXCoord := getHeaderToXCoord(rows[0])
+	usernameXCoord := headerToXCoord[input.UsernameHeader]
+	passwordXCoord := headerToXCoord[input.PasswordHeader]
+	rowOffset := input.rowOffset
+
+	for _, row := range rows[rowOffset:] {
+		// check if row is empty
+		if row == nil {
+			continue
+		}
+		users[row[usernameXCoord]] = row[passwordXCoord]
+	}
+
+	return users, nil
+}
+
+func getPasswordManagerUsers(f *excelize.File, input *Input) (usersToPassword map[string]string, usersToRow map[string]int, err error) {
```

Comment:
I can't find your comment to reply to, but how about:

```
type PasswordManagerUserDetails struct{
  Password string
  Row int
}
```

and then
```
func getPasswordManagerUsers(f *excelize.File, input *Input) (users map[string]PasswordManagerUserDetails, err error) {
```

This makes it clear that users always come with a both a row and a password.

Reply:
Yup. I already implemented this struct, per our discussion.